### DISCUSSION TITLE
Will Prisma create field for entity that is connected or it will be left as null?

### DISCUSSION THREAD

**Discussion by tonyabracadabra** [prisma/prisma#2816] [Q&A] [answered]
I am trying to add a new `Course` to `User`, `User` and `Course` is connected via `instructorId` in the following schema I defined

```
model User {
  id        ID   
  password  String
  courses   Course[]
}

model Course {
  id ID
  name String
  description String
  instructor User @relation(fields: [instructorId], references: [id])
  instructorId String
}
```
My course creation is written as
```
async createCourse(name: string, description: string, instructorId: string) {
    return this.prisma.course.create({
      data: {
        name: name,
        description: description,
        instructor: {
          connect: {
            id: instructorId
          }
        },
      },
    });
  }
```
This function does not give me any errors, which is fine, but when I look back into the database (prisma studio), I found that though I have the `userId` linked to the course I newly created, it does not help creating the `instructor` field in `Course` and is showing as `null`

**Comment by ryands17** [comment, marked as answer]
Hey @tonyabracadabra :wave: 

First of all, your schema is incorrect. It should be:

```prisma
model User {
  id       String      @id @default(uuid())
  password String
  courses  Course[]
}

model Course {
  id           String    @id @default(uuid())
  name         String
  description  String
  instructor   User   @relation(fields: [instructorId], references: [id])
  instructorId String
}
```

And I have run the above query and studio is showing the connected values for me. Could you check by querying back for the same record and see if the relations are showing?

**Reply by tonyabracadabra** [reply]
Hey Ryan thanks for answering this!

I've tried your way of writing schema and it doesn't really makes a difference for me, I was using `ID` because I have defined a type `type ID = String @id @default(uuid())` in the front of the file so I guess that is expected that nothing will be changed.

It is okay for me to create course with GraphQL like this and it is giving me legitimate result like I expected

```graphql
mutation {
  createCourse(addCourse: {name: "xxx", description: "xxxxx"}) {
    id
    name
    description
  }
}
```

but when I add an additional return field `instructor`, 
```graphql
mutation {
  createCourse(addCourse: {name: "xxx", description: "xxxxx"}) {
    id
    name
    description
    instructor {
        id
    }
  }
}
```
since that field is empty/null, it gives me error 
`"message": "Cannot return null for non-nullable field Course.instructor."`

And when I opened up prisma studio page, I can't load `course` page because I was getting 

```Error in Prisma Client request: 

Error: 
Invalid `prisma.course.findMany()` invocation:
```

I think that's because of the same reason that `instructor` field is null.

**Reply by tonyabracadabra** [reply]
That being said, for my case I can see list of `courses` in `instructor` table on prisma studio, but not the other way around, the other direction only has `course.instructorId`, but `course.instructor` is null.

**Reply by ryands17** [reply]
Could you share your schema then if possible?

**Reply by tonyabracadabra** [reply]
I've changed the schema to the form of what you have suggested. 
This is what I can see from users page
![image](https://user-images.githubusercontent.com/6690727/85264059-f6c2b800-b424-11ea-83d8-021b3381c83a.png)

This is what I can see after I clicked into `coursesTaught` in that row
![image](https://user-images.githubusercontent.com/6690727/85264200-2f629180-b425-11ea-8bcd-4c6231365de6.png)

**Reply by ryands17** [reply]
@tonyabracadabra I ran the below query:

```ts
const user = await prisma.user.create({
    data: {
      password: 'password1',
    },
  })
  await prisma.course.create({
    data: {
      description: 'desc 1',
      name: 'Course 1',
      instructor: {
        connect: {
          id: user.id,
        },
      },
    },
  })
```

And here's the Studio output:
![image](https://user-images.githubusercontent.com/19697099/85270018-573e0d80-b496-11ea-9c0c-e4f055f8e185.png)

So I cannot reproduce your issue. It's best if you create a bug report [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=).

**Reply by tonyabracadabra** [reply]
Thanks Ryan, were you using postgres as backend for the query?

**Reply by ryands17** [reply]
I have tried on Postgres and SQLite and both are showing the connected relation in Studio. I am using version 2.0.1 of  `@prisma/client` and `@prisma/cli`. Is that the same version you're on?

**Reply by tonyabracadabra** [reply]
I think I figured that out. I didn't do the @ResolveField for `instructor` field and I guess that is one crucial step to it.

```typescript
@ResolveField('instructor')
async instructor(@Parent() post: Course) {
  return this.prisma.course.findOne({ where: { id: post.id } }).instructor();
}
```

I think maybe there is anything that we can improve it by automatically generate the @ResolveField function?

**Comment by ludralph** [comment]
Hi @tonyabracadabra,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
